test/Conversion/ExportVerilog/hw-dialect.mlir

```diff
@@ -1392,12 +1401,32 @@ hw.module @DontInlineAggregateConstantIntoPorts() -> () {
 
 // CHECK-LABEL: module FooA(
 // CHECK-NEXT:    input union packed {logic [15:0] a; struct packed {logic [9:0] b; logic [5:0] __post_padding_b;} b;} test
+// CHECK-NEXT:    output [15:0] a,
+// CHECK-NEXT:    output [9:0] b
 // CHECK-NEXT:  );
+// CHECK-EMPTY:
+// CHECK-NEXT:    assign a = test.a;
+// CHECK-NEXT:    assign b = test.b.b;
+// CHECK-NEXT:  endmodule
 !unionA = !hw.union<a: i16, b: i10>
-hw.module @FooA(%test: !unionA) {}
+hw.module @FooA(%test: !unionA) -> (a: i16, b: i10) {
+  %0 = hw.union_extract %test["a"] : !unionA
+  %1 = hw.union_extract %test["b"] : !unionA
+  hw.output %0, %1 : i16, i10
+}
 
 // CHECK-LABEL: module FooB(
-// CHECK-NEXT:    input union packed {logic [15:0] a; struct packed {logic [1:0] __pre_padding_b; logic [9:0] b; logic [3:0] __post_padding_b;} b;} test
+// CHECK-NEXT:    input union packed {logic [15:0] a; struct packed {logic [1:0] __pre_padding_b; logic [13:0] b;} b;} test,
+// CHECK-NEXT:    output [15:0] a,
+// CHECK-NEXT:    output [13:0] b
 // CHECK-NEXT:  );
-!unionB = !hw.union<a: i16, b: i10 offset 2>
-hw.module @FooB(%test: !unionB) {}
+// CHECK-EMPTY:
+// CHECK-NEXT:    assign a = test.a;
+// CHECK-NEXT:    assign b = test.b.b;
+// CHECK-NEXT:  endmodule
+!unionB = !hw.union<a: i16, b: i14 offset 2>
+hw.module @FooB(%test: !unionB) -> (a: i16, b: i14) {
+  %0 = hw.union_extract %test["a"] : !unionB
+  %1 = hw.union_extract %test["b"] : !unionB
+  hw.output %0, %1 : i16, i14
+}
```

Comment:
I changed this test to only have pre_padding, so now we have a test with only post_padding and a test with only pre_padding.